### PR TITLE
Add dimension check in tensordot

### LINKED ISSUE #91589: Unknown error when running torch.tensordot on torch 1.12.0 and 1.13.1
### 🐛 Describe the bug

I get unknown error when running torch.tensordot with the following input combinations:

```
import torch
import numpy as np
arg_1_tensor = torch.rand([0, 0, 1, 2, 3, 0], dtype=torch.float64)
arg_1 = arg_1_tensor.clone()
arg_2_tensor = torch.rand([1, 2, 3, 0], dtype=torch.float64)
arg_2 = arg_2_tensor.clone()
arg_3 = 95063276620
try:
  res = torch.tensordot(arg_1,arg_2,dims=arg_3,)
except Exception as e:
  print("Error:"+str(e))
```

The log message is:
```
Error:

```

### Versions

PyTorch version: 1.12.0
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.9.15 (main, Nov 24 2022, 14:31:59) [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-56-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to:
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1660 Ti
Nvidia driver version: 525.60.11
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.5
[pip3] torch==1.12.0
[pip3] torchaudio==0.12.0
[pip3] torchvision==0.13.0
[conda] blas 1.0 mkl
[conda] cudatoolkit 11.3.1 h2bc3f7f_2
[conda] ffmpeg 4.3 hf484d3e_0 pytorch
[conda] mkl 2021.4.0 h06a4308_640
[conda] mkl-service 2.4.0 py39h7f8727e_0
[conda] mkl_fft 1.3.1 py39hd3c417c_0
[conda] mkl_random 1.2.2 py39h51133e4_0
[conda] numpy 1.23.5 py39h14f4228_0
[conda] numpy-base 1.23.5 py39h31eccc5_0
[conda] pytorch 1.12.0 py3.9_cuda11.3_cudnn8.3.2_0 pytorch
[conda] pytorch-mutex 1.0 cuda pytorch
[conda] torchaudio 0.12.0 py39_cu113 pytorch
[conda] torchvision 0.13.0 py39_cu113 pytorch

### PR DESCRIPTION
**Pull request by yanbing-j**
This PR is to add dimension check in tensordot. The expected dimension should be smaller than `dim_a` or `dim_b`.
Fix #91589

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: torch/functional.py [hunk 1]
```diff
@@ -1093,6 +1093,11 @@ def tensordot(a, b, dims=2, out: Optional[torch.Tensor] = None):  # noqa: F811
     if isinstance(dims, int):
         if dims < 0:
             raise RuntimeError(f"tensordot expects dims >= 0, but got dims={dims}")
+        ndim_a = a.dim()
+        ndim_b = b.dim()
+        expected_ndim = ndim_a if ndim_a <= ndim_b else ndim_b
+        if dims > expected_ndim:
```
Comment (jgong5) [inline torch/functional.py:1099]: `if dim > min(a.dim(), b.dim())`?

Comment (yanbing-j) [reply torch/functional.py]: Yes. Done.

**Review by jgong5 — APPROVED** [review]

**Review by albanD — APPROVED** [review]
Could you add a test?

**Comment by yanbing-j** [comment]
Sure. Done.
